sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/webapp/AADOAuth2AuthorizationRequestResolver.java

```diff
@@ -61,7 +61,7 @@ private OAuth2AuthorizationRequest addClaims(HttpServletRequest httpServletReque
                 additionalParameters.put(Constants.CLAIMS, conditionalAccessPolicyClaims);
             }
         }
-        if (properties != null) {
+        if (properties != null && properties.getAuthenticateAdditionalParameters() != null) {
```

Comment:
Hi, @stliu , please help to review this PR.

And it's error-prone to write code like this.
That's why I'd like to use `Optional` to avoid NPE.

Of course, you are the boss, I won't use `Optional` in this project until you allow me to.